mlir/tools/mlir-miopen-lib/mlir-miopen-lib.cpp

```diff
@@ -259,10 +261,11 @@ extern "C" MiirStatus miirLowerTuningParams(MiirHandle mlirHandle) {
   PassManager pm(module.getContext(), PassManager::Nesting::Implicit);
 
   // Passes for lowering MIOpen dialect.
-  pm.addPass(mlir::miopen::createLowerMIOpenOpsStep1Pass());
-  pm.addPass(mlir::miopen::createAffineTransformPass());
   pm.addPass(
       mlir::miopen::createAffixTuningParametersPass(0, 0, handle->perfConfig));
+  pm.addPass(mlir::miopen::createLowerMIOpenOpsStep1Pass());
+  pm.addPass(mlir::miopen::createAffineTransformPass());
+  pm.addPass(mlir::miopen::createLowerMIOpenOpsStep2Pass());
```

Comment:
Now we can actually need to lower more pipelines to be able to tell if it is going to succeed? From MIOpen perspective they want to keep the applicability check absolutely minimal because that was on the critical pass. Is there any way to remain the same amount of lowering as before.

Also I'd kick off a MIOpen integration round of test on this PR when it is ready just to make sure. Those changes will have impact on it.

Reply:
BTW, once you've done that, could you add a few test cases (Doesn't have to be in this PR) that make sure certain configs to make either of the above passes to fail. This is just to make sure a `signalPassFailure` in either of those passes can be propagated correctly to API as a `MIIR_BUILD_FAILURE`

Reply:
With upcoming KPACK logic, we need step2 to tell whether we have enough data to compute along GemmK dimension.
And step2 actually doesn't take too much time, compared with LLVM middle-end (opt) or backend (llc).

Additional tests shall likely be added in the KPACK PR.

Reply:
> And step2 actually doesn't take too much time

Sounds good.

> Additional tests shall likely be added in the KPACK PR.

Is there a way to track it in a ticket? This is pretty important because I want to make sure we reject incorrect configs properly, otherwise bad things can happen from our client.

Reply:
It will be covered in https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/199

Reply:
Documented [here](https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/199#issuecomment-928257850).